hangman/lib/game/tally.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Hangman.Tally do
+
+  defstruct(
+    game_state: :initializing,
+    turns_left: 7,
+    letters: [],
+    used: [],
+    last_guess: ''
+  )
+
+  #returns the current game-card
+  def tally(game) do
+    determine_tally(game, game.letters, [])
+  end
+
+  def determine_tally(game, [h1 | t1], final) do
+    determine_tally(game, t1, check(Enum.member?(game.used, h1), h1, final))
```

Comment:
This code definitely works, but it seems hard to understand, partly because the naming of things could be better (`check` doesn't really mean much), partly because some lines (like 17) have too much going on, and partly because you might be doing more work than you need to. For example, if the incoming state is `already_used` there's no need to perform and won or lost checks, if the state is `bad_guess` then you only check for lost, and so on